modules/add-pod-namespace-to-sensor-and-admission-control.adoc

```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * upgrade/upgrade-from-45.adoc
+:_mod-docs-content-type: PROCEDURE
+[id="add-pod-namespace-to-sensor-and-admission-control_{context}"]
+= Add `POD_NAMESPACE` to `sensor` and `admission-control` deployments
+
+[role="_abstract"]
+When upgrading to a version >= 4.6 from a version < 4.6, you must patch the sensor and admission-control deployments  to set the `POD_NAMESPACE` environment variable.
+
+[NOTE]
+====
+If you are using Kubernetes, use `kubectl` instead of `oc` for the commands listed in this procedure.
+====
+
+.Procedure
+
+. Patch sensor to ensure `POD_NAMESPACE` is set:
++
+[source,terminal,subs=attributes+]
+----
+$ oc -n stackrox patch deployment sensor --type=json -p '[{"op":"add","path":"/spec/template/spec/containers/0/env/-","value":{"name":"POD_NAMESPACE","valueFrom":{"fieldRef":{"fieldPath":"metadata.namespace"}}}}]'
```

Comment:
`-n stackrox` - I think, we should assume that this may be different in general - are we using simply `stackrox` constant in other places like this?

Reply:
I've looked into it and there is no way to do this purely with a `JSON` patch. There's the `ADD` operation which will create duplicates, and the `REPLACE` operation which will fail if the item is not yet present. 

However a simple bash query for `POD_NAMESPACE` inside the yaml file should do. The only downside is that the command now requires the user to have grep installed.

Reply:
> However a simple bash query for POD_NAMESPACE inside the yaml file should do. The only downside is that the command now requires the user to have grep installed.

Not sure how I feel about the bash script. On one hand, I think it's better as it would ensure the variable is not set multiple times. On the other hand, having just
```
oc -n stackrox patch deployment admission-control --type=json -p '[{"op":"add","path":"/spec/template/spec/containers/0/env/-","value":{"name":"POD_NAMESPACE","valueFrom":{"fieldRef":{"fieldPath":"metadata.namespace"}}}}]'
```
looks cleaner to me and more readable.

Reply:
Well, setting the env multiple times in the .yaml file does not break the file but it's ugly.

A third option I could offer would be adding a script that does all the patching and then just put a curl command to the script into the docs, but I don't like this so much when it's oneliner commands.